app/models/spawn_job.rb

```diff
@@ -8,6 +8,11 @@
 #
 class SpawnJob < EventAction
   def run!
-    Job.new(job_type: job_type, worker: user.workers.assignment_order.first, status: "DISPATCHED")
+    worker = user.workers.assignment_order.first
+    if worker
+      Job.create(job_type: job_type, worker: worker, status: "DISPATCHED")
+    else
+      ScheduledWorker.perform_at(Worker::RETRY_TIME.from_now, id)
```

Comment:
You're passing it the `EventAction` id here, not the `EventReceiver` id. I'm not sure this will work the way you want it to, since the `ScheduledWorker` job find the event receiver matching the id you pass in.

```ruby
# frozen_string_literal: true

class ScheduledWorker
  include Sidekiq::Worker
  attr_reader :receiver

  def perform(id)
    @receiver = Receiver.find_by(id)
    return unless receiver
    schedule_job
  end

  private

  def schedule_job
    if worker
      @job = Job.new(job_type: receiver.job_type, worker: worker, status: "DISPATCHED")
    else
      ScheduledWorker.perform_at(Worker::RETRY_TIME.from_now, id)
    end
  end

  def worker
    receiver
      &.user
      &.workers
      &.assignment_order
      &.first
  end
end
```

Reply:
ah true good catch